localstack-core/localstack/services/lambda_/api_utils.py

```diff
@@ -185,14 +246,21 @@ def get_function_name(function_arn_or_name: str, context: RequestContext) -> str
     return name
 
 
-def function_locators_from_arn(arn: str) -> tuple[str, str | None, str | None, str | None]:
+def function_locators_from_arn(arn: str) -> tuple[str | None, str | None, str | None, str | None]:
     """
     Takes a full or partial arn, or a name
 
     :param arn: Given arn (or name)
     :return: tuple with (name, qualifier, account, region). Qualifier and region are none if missing
     """
-    return FUNCTION_NAME_REGEX.match(arn).group("name", "qualifier", "account", "region")
+
+    def _extract_named_groups(name, qualifier, account, region):
+        return name, qualifier, account, region
+
+    if matched := FUNCTION_NAME_REGEX.match(arn):
+        return _extract_named_groups(**matched.groupdict())
+
+    return 4 * (None,)
```

Comment:
Why do we need the internal function for this? Is there an advantage for this to using `matched.group(names...)`?

I would also prefer `None, None, None, None` over `4 * (None, )`, as I think it is more readable and clearer on the intent.

Reply:
> Why do we need the internal function for this? 

Previously I was doing qualifier and name validation within that internal function. It made it a lot cleaner since we could just unpack the dict args to that internal func and easily validate individual members. Since validation takes place elsewhere I agree this is unnecessary.

i.e 
```python
    def _extract_named_groups(name, qualifier, account, region):
        validate_name_and_qualifier(name, qualifier) # This would raise an exception
        return name, qualifier, account, region

    if matched := FUNCTION_NAME_REGEX.match(arn):
        return _extract_named_groups(**matched.groupdict())
```

> I would also prefer `None, None, None, None`

Fair. I think the current approach takes "pythonic" to the extreme lol. Will change.